tests/test_application.py

```diff
@@ -194,3 +194,37 @@ def test_acquire_token_silent(self):
         self.assertNotEqual(None, at)
         self.assertEqual(self.access_token, at.get('access_token'))
 
+class TestClientApplicationForSuberrors(unittest.TestCase):
+
+    @classmethod
+    def setUp(self):
+        self.authority_url = "https://login.microsoftonline.com/common"
+        self.authority = msal.authority.Authority(self.authority_url)
+        self.scopes = ["s1", "s2"]
+        self.uid = "my_uid"
+        self.utid = "my_utid"
+        self.account = {"home_account_id": "{}.{}".format(self.uid, self.utid)}
+        self.frt = "what the frt"
+        self.cache = msal.SerializableTokenCache()
+        self.client_id = "my_app"
+        self.access_token = "access token"
+        self.cache.add({  # Pre-populate a FRT
+            "client_id": self.client_id,
+            "scope": self.scopes,
+            "token_endpoint": "{}/oauth2/v2.0/token".format(self.authority_url),
+            "response": TokenCacheTestCase.build_response(
+                access_token=self.access_token,
+                uid=self.uid, utid=self.utid, refresh_token=self.frt),
+        })  # The add(...) helper populates correct home_account_id for future searching
+
+    def test_error_response(self):
+        app = ClientApplication(
+            self.client_id,
+            authority=self.authority_url, token_cache=self.cache)
+        def tester(url, data=None, **kwargs):
+            return Mock(status_code=200, json=Mock(return_value={
+                "error": "invalid_grant",
+                "error_description": "Was issued to another client",
```

Comment:
Perhaps we can remove this line? I have not tested it though.

```suggestion
```

Reply:
This cant be removed as error description is needed when logger is trying to write debug logs

Reply:
Removing this will break the test as the logger is using error description value in the library

Reply:
Good to know. Then we should change our library instead, to remove our reliance on the error_description. I came up with [a PR](https://github.com/AzureAD/microsoft-authentication-library-for-python/pull/140) just now, please help review. After that, we can remove this line here.